### PR TITLE
MGMT-11262: Parse `/etc/kubernetes/kubeconfig` in day-2 ignition to disable DNS validations

### PR DESCRIPTION
**Pull request by omertuc**
When the ignition downloaded by the day-2 worker discovery indicates
that kubelet is connecting to the day-1 cluster using an IP address
rather than a domain name, we can safely disable DNS validations for
that worker in day-2.

This is the case in Hypershift.

The agent was modified to include /etc/kubernetes/kubeconfig in the
day-2 ignition to allow this, because previously it was omitted for size
concerns

This commit modifies the service to look at this file that is now
included by the agent and parse it to choose whether DNS validations
need to be enabled or disabled

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
    - Asked @eranco74 to test this manually
- [ ] No tests needed

## Assignees

/cc @eranco74
/cc @tsorya

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by omertuc** [comment]
```
    {
      "id": "api-domain-name-resolved-correctly",
      "status": "success",
      "message": "Domain name resolution for the api.test-infra-cluster-4b99dadd.redhat.com domain was successful or not required"
    },
    {
      "id": "api-int-domain-name-resolved-correctly",
      "status": "success",
      "message": "Domain name resolution for the api-int.test-infra-cluster-4b99dadd.redhat.com domain was successful or not required"
    },
    {
      "id": "apps-domain-name-resolved-correctly",
      "status": "success",
      "message": "Domain name resolution for the *.apps.test-infra-cluster-4b99dadd.redhat.com domain was successful or not required"
    },
```

looking good

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: internal/host/validator.go [hunk 1]
```diff
@@ -1270,6 +1298,50 @@ func ignitionContainsManagedNetworkingFiles(config *ignition_types.Config) bool
...
+	}
+	type Kubeconfig struct {
+		Clusters []KubeconfigCluster `json:"clusters"`
+	}
+	var kubeconfig Kubeconfig
+
+	err = yaml.Unmarshal(kubeconfigBytes, &kubeconfig)
+	if err != nil {
+		return false
+	}
```
Comment (eranco74) [inline internal/host/validator.go:1330]: Why not use [KubectlConfig](https://github.com/kubernetes/kops/blob/v1.24.0/pkg/kubeconfig/config.go#L19)

**Comment by eranco74** [comment]
> /test edge-e2e-metal-assisted-capi

https://github.com/openshift/assisted-service/pull/4174

File: internal/host/validator.go (hunk 1 above)
Comment (omertuc) [reply internal/host/validator.go:1330]: Seems overkill to depend on kops for something so simple

**Comment by eranco74** [comment]
The test looks good.
/unhold

**Pull request merged**